packages/db/src/resources/contracts.ts

```diff
@@ -106,6 +166,18 @@ export const contracts: Definition<"contracts"> = {
           return result;
         }
       }
+    },
+    ABI: {
+      entries: {
+        resolve({ json }) {
+          return normalize(JSON.parse(json));
+        }
+      }
+    },
+    Entry: {
+      __resolveType(obj) {
+        return `${pascalCase(obj.type)}Entry`;
```

Comment:
What's going on here? I'm not sure I understand.

Reply:
Since `Entry` is polymorphic, GraphQL needs a discriminator, so we use `type` to construct the name for the correct Entry type (`FunctionEntry`, etc.)

Reply:
I see! Kind of an odd way of doing it, but I guess that works!  So internally, the discriminator will be `"FunctionEntry"` rather than `"function"`, huh?  I guess that works.